server/app/middleware/authenticate.js

```diff
@@ -0,0 +1,49 @@
+import jwt from 'jsonwebtoken';
+import db from '../../models';
+
+/**
+ * Class to implement authentication middlewares
+ */
+class Authenticate {
+  /**
+   * Method to authenticate a user before proceeding
+   * to protected routes
+   * @param {Object} req - The request Object
+   * @param {Object} res - The response Object
+   * @param {Function} next - Function call to move to the next middleware
+   * or endpoint controller
+   * @return {Void} - Returns void
+   */
+  static auth(req, res, next) {
+    const token = req.headers['x-access-token'] || req.body.token;
+    if (token) {
+      jwt.verify(token, 'secret', (error, decoded) => {
```

Comment:
You may want to use environment variable for your `secret` here instead of hard coding it here.